.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,12 +1,22 @@
-<!--
-Please follow the naming conventions in the "guidelines for contributing" link above or the pull request may be rejected.
+Fixes #
 
-Also, do consider outlining the solution taken as doing so will likely help in the reviewing process
-(e.g., when the pull request involves non-trivial changes)
--->
+**PR Checklist**
 
-Fixes #
+<!-- Remove this portion after you have made the checks. -->
+
+Have you:
+- [ ] Read and understood [our PR guideline (steps 4-6)](https://github.com/TEAMMATES/teammates/blob/master/docs/process.md#step-4-submit-a-pr)?
+  - [ ] Added the issue number to the "Fixes" keyword above?
+  - [ ] Titled the PR as specified in the abovementioned document?
+- [ ] Made your changes on a branch other than `master`?
+- [ ] Used good commit messages?
```

Comment:
> instead of spending time on the code quality which ultimately translates to software quality, time is spent on organizing commits and reviewing them.

Unfortunately, in our case, much time is wasted deciphering ill-organised commits and poor commit messages instead of actual reviewing. We don't need to adopt the entire set of guidelines immediately and strictly, since it is only for the reviewers' benefit at the moment. However, what we can (and should) do is lay down some concrete guidelines that contributors are expected to follow for their commits. I believe many good practices are already informally in use by maintainers anyway, what we need is for the new contributors to follow them as well.